### ISSUE TITLE
Incorrect Scaling of Quantiles in Validation Loss

### ISSUE THREAD

**Issue by carlocav** [Nixtla/neuralforecast#1431]
### What happened + What you expected to happen

While training a DeepAR model, I noticed unexpected values in the validation loss and identified the following issue.

When the training loss is DistributionLoss and the validation loss is sCRPS, MQLoss, or HuberMQLoss, quantiles are computed from the predictive distribution:

https://github.com/Nixtla/neuralforecast/blob/3bb21b625606f13116adb44bf817b45db822b087/neuralforecast/common/_base_model.py#L1002

Quantiles are then evaluated using these loss functions, however, they are scaled back, which is incorrect, as they appear, or should, to be already computed from samples generated from a distribution in the original scale.

Therefore, the following line should not be executed in this case:

https://github.com/Nixtla/neuralforecast/blob/3bb21b625606f13116adb44bf817b45db822b087/neuralforecast/common/_base_model.py#L1014

### Versions / Dependencies

neuralforecast 3.1.2
Python 3.12

### Reproduction script

For example, in the DeepAR usage example, the validation loss is computed incorrectly.
```
import pandas as pd
import matplotlib.pyplot as plt

from neuralforecast import NeuralForecast
from neuralforecast.models import DeepAR
from neuralforecast.losses.pytorch import DistributionLoss, MQLoss
from neuralforecast.utils import AirPassengersPanel, AirPassengersStatic
Y_train_df = AirPassengersPanel[AirPassengersPanel.ds<AirPassengersPanel['ds'].values[-12]] # 132 train
Y_test_df = AirPassengersPanel[AirPassengersPanel.ds>=AirPassengersPanel['ds'].values[-12]].reset_index(drop=True) # 12 test

nf = NeuralForecast(
    models=[DeepAR(h=12,
                   input_size=24,
                   lstm_n_layers=1,
                   trajectory_samples=100,
                   loss=DistributionLoss(distribution='StudentT', level=[80, 90], return_params=True),
                   valid_loss=MQLoss(level=[80, 90]),
                   learning_rate=0.005,
                   stat_exog_list=['airline1'],
                   futr_exog_list=['trend'],
                   max_steps=100,
                   val_check_steps=10,
                   early_stop_patience_steps=-1,
                   scaler_type='standard',
                   enable_progress_bar=True,
                   ),
    ],
    freq='ME'
)
nf.fit(df=Y_train_df, static_df=AirPassengersStatic, val_size=12)
Y_hat_df = nf.predict(futr_df=Y_test_df)
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.
